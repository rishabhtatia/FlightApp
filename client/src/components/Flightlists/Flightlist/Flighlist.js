import React from 'react';
import PropTypes from 'prop-types';
import styles from './Flightlist.module.css';
import clsx from 'clsx';

const Flightlist = ({ data, multiple }) => {
  return (
    <div
      className={clsx(styles.flightList, styles.Blackborder && !data.multiple)}
    >
      <div className="card">
        <div className="card-body">
          <div className="row">
            <div className="col-sm">
              <h4>{data.name}</h4>
              <span>{data.flightNo}</span>
            </div>
            <div className="col-sm">
              <h4>{data.departureTime}</h4>
              <span>{data.origin}</span>
            </div>
            <div className="col-sm">
              <h4>{data.arrivalTime}</h4>
              <span>{data.destination}</span>
            </div>
            <div className="col-sm">
              <h4>{data.journeyTime}</h4>
              <span>Non Stop</span>
            </div>
            <div className="col-sm">
              {!multiple && (
                <h4 className={styles.redColor}>
                  {new Intl.NumberFormat('en-IN', {
                    style: 'currency',
                    currency: 'INR',
                  }).format(data.price)}
                </h4>
              )}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

Flightlist.defaultpropTypes = {
  data: {},
};

Flightlist.propTypes = {
  data: PropTypes.any,
  multiple: PropTypes.bool,
};

export default Flightlist;
