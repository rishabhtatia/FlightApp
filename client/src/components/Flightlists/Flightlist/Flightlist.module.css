.flightList {
  margin: 5px;
}
.blackBorder {
  border: 1px solid black;
}
.redColor {
  color: red;
}
