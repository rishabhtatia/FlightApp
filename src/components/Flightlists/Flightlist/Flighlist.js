import React from 'react';
import moment from 'moment';
import styles from './Flightlist.module.css';

const FLIGHTLIST = (props) => {
  const data = props.data;
  const journeyTimestart = moment(data.date + ' ' + data.departureTime);
  const journeyTimeend = moment(data.date + ' ' + data.arrivalTime);  
  let totaljourneyTime = journeyTimeend.diff(
    journeyTimestart,
    'DD/MM/YYYY HH:mm:ss'
  );  
  totaljourneyTime = moment.duration(totaljourneyTime);    
  return (
    <div className={styles.flightList + " " + (!props.multiple ? styles.border: '')}>
      <div className="card">
        <div className="card-body">
          <div className="row">
            <div className="col-sm">
              <h4>{data.name}</h4>
              <span>{data.flightNo}</span>
            </div>
            <div className="col-sm">
              <h4>{data.departureTime}</h4>
              <span>{data.origin}</span>
            </div>
            <div className="col-sm">
              <h4>{data.arrivalTime}</h4>
              <span>{data.destination}</span>
            </div>
            <div className="col-sm">
              <h4>{totaljourneyTime._data.hours + 'h:' + totaljourneyTime._data.minutes + 'm'}</h4>
              <span>Non Stop</span>
            </div>
            <div className="col-sm">
              {!props.multiple && (
                <h4 className={styles.price}>
                  {new Intl.NumberFormat('en-IN', {
                    style: 'currency',
                    currency: 'INR',
                  }).format(data.price)}
                </h4>
              )}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default FLIGHTLIST;
