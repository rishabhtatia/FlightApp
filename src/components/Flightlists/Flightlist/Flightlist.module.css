.flightList {
  margin: 5px;
}
.border {
  border: 1px solid black;
}
.price {
  color: red;
}
