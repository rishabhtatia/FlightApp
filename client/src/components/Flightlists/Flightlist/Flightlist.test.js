import React from 'react';
import renderer from 'react-test-renderer';

import Flightlist from './Flighlist';

it('renders when there is one item', () => {
  const item = {
    arrivalTime: '10:20',
    date: '2020/11/01',
    departureTime: '8:10',
    destination: 'Delhi (DEL)',
    flightNo: 'AI-104',
    name: 'Air India',
    origin: 'Pune (PNQ)',
    price: 4681,
  };
  const tree = renderer.create(<Flightlist data={item} />).toJSON();
  expect(tree).toMatchSnapshot();
});
