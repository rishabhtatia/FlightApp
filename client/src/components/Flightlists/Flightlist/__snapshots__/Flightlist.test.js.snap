// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there is one item 1`] = `
<div
  className="flightList"
>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <div
        className="row"
      >
        <div
          className="col-sm"
        >
          <h4>
            Air India
          </h4>
          <span>
            AI-104
          </span>
        </div>
        <div
          className="col-sm"
        >
          <h4>
            8:10
          </h4>
          <span>
            Pune (PNQ)
          </span>
        </div>
        <div
          className="col-sm"
        >
          <h4>
            10:20
          </h4>
          <span>
            Delhi (DEL)
          </span>
        </div>
        <div
          className="col-sm"
        >
          <h4>
            2h:10m
          </h4>
          <span>
            Non Stop
          </span>
        </div>
        <div
          className="col-sm"
        >
          <h4
            className="redColor"
          >
            â‚¹4,681.00
          </h4>
        </div>
      </div>
    </div>
  </div>
</div>
`;
